### DISCUSSION TITLE
Can I access a record created in a nested write within the same query?

### DISCUSSION THREAD

**Discussion by wkd-kapsule** [prisma/prisma#23004] [Q&A]
### Question

Is there a way to do something like this in a single query?
Or am I bound to do the nested write first and then add the id in a second update query?

### How to reproduce (optional)

``` typescript
await prisma.space.update({
  where: {id: space.id},
  data: {
    members: {
      create: {
        email: dbUser.email,
        name: dbUser.name,
        image: dbUser.image,
      },
    },
    membersId: {
      push: {createdMember.id} //createdMember would be the member created just above.
    }
  },
})
```

**Comment by ludralph** [comment]
Hi @kapsule-studio 👋 

Thank you for raising this question.

You cannot access a record created in a nested write within the same query.

In your case, you might need to perform the nested write first and then use the returned ID in a second update query. Here's an example of how you might do this:
```js
// First, create the member
const createdMember = await prisma.space.update({
  where: {id: space.id},
  data: {
    members: {
      create: {
        email: dbUser.email,
        name: dbUser.name,
        image: dbUser.image,
      },
    },
  },
})

// Then, update the membersId field
await prisma.space.update({
  where: {id: space.id},
  data: {
    membersId: {
      push: createdMember.id
    }
  },
})
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
